vars/capsulePipelineWrapper.groovy

```diff
@@ -1,245 +1,485 @@
+            --nodeset-group-name validators \
+            --type vega \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/tendermint.common.tmpl.toml' \
+            --nodeset-group-name validators \
+            --type tendermint \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/vega.full.tmpl.toml' \
+            --nodeset-group-name full \
+            --type vega \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/tendermint.common.tmpl.toml' \
+            --nodeset-group-name full \
+            --type tendermint \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/data_node.full.tmpl.toml' \
+            --nodeset-group-name full \
+            --type data-node \
+            --with-merge """ + additionalFlags
+}
+
+void getReleasedBinary(String ghCredentialsID, String binType, String version) {
+  withGHCLI('credentialsId': ghCredentialsID) {
+        sh 'gh release --repo vegaprotocol/vega download ' + version + ' --pattern "' + binType + '-linux*"'
+  }
+
+  sh 'mv vega-linux-amd64 bin/' + binType
+  sh 'chmod +x bin/' + binType
+  sh binType + ' version'
+}
+
+void compileCoreBinary(String ghCredentialsID, String binType, String version, String outDir) {
+  Map packagesMapping = [
+        'vega': 'cmd/vega/',
+        'vegawallet': 'cmd/vegawallet/',
+        'data-node': 'cmd/data-node/'
+    ]
+
+  if (!packagesMapping.containsKey(binType)) {
+    error('[compileBinary] invalid binary type. got: "' + binType + '"')
+  }
+
+  gitClone([
+      url: 'git@github.com:vegaprotocol/vegacapsule.git',
+      branch: 'main',
+      directory: version,
+      credentialsId: ghCredentialsID
+  ])
+  sh label: 'Building ' + binType,
+        script: 'go mod vendor && go build -o ' + outDir + '/' + binType + ' ' + packagesMapping[binType]
+}
+
+void downloadS3Binary(String s3Path, String binName, String outDir) {
+  sh label: 'Download the ' + binName + ' from the S3',
+       script: 'aws s3 cp --only-show-errors --no-progress "' + s3Path + '" "' + outDir + '/' + binName + '"'
+  sh label: 'Set permissions on the ' + binName + ' binary',
+       script: 'chmod a+x "' + outDir + '/' + binName + '"'
+}
+
+boolean isS3Link(String path) {
+    /* groovylint-disable-next-line DuplicateNumberLiteral */
+  return path.length() > 0 && path[0..4] == 's3://'
+}
+
+void veganetSh(String network, String command, String sshCredentialsId, String dockerCredentialsId) {
```

Comment:
Let's extract this method to external file and use it together with pipelineDeploy to reduce copy paste

Reply:
fixing